miso-web/src/main/webapp/scripts/hot.js

```diff
@@ -956,58 +992,58 @@ var HotUtils = {
       }
     }, ];
   },
-  
-  makeParents: function(slug, parentCategories){
-  	return HotUtils.makeRelations(slug + '/parents', 'Parents', parentCategories, true);
+
+  makeParents: function(slug, parentCategories) {
+    return HotUtils.makeRelations(slug + '/parents', 'Parents', parentCategories, true);
   },
-  
-  makeChildren: function(slug, childCategories){
-  	return HotUtils.makeRelations(slug + '/children', 'Children', childCategories, false);
+
+  makeChildren: function(slug, childCategories) {
+    return HotUtils.makeRelations(slug + '/children', 'Children', childCategories, false);
   },
-  
+
   makeRelations: function(slug, relationship, relationCategories, useParentBound) {
 
     return {
       name: relationship,
       action: function(items) {
-      	function makeCategoriesFilter(items){
-      		if(!Constants.isDetailedSample){
-      			return function(category){
-      				if(useParentBound){
-      					return category.index >= Constants.sampleCategories.length;
-      				}
-      				return category.index > Constants.sampleCategories.length;
-      			};
-      		}
-      		var childBound = Constants.sampleCategories.length - 1;
-      		var parentBound = 0;
-      		for(sample in items){
-      			if(items[sample].sampleClassId === undefined){
-      				parentBound = Constants.sampleCategories.length - 1;
-      				continue;
-      			}
-      			var index = Constants.sampleCategories.indexOf(Constants.sampleClasses.find(function(sampleClass){
-      				return sampleClass.id == items[sample].sampleClassId;
-      			}).sampleCategory);
-      			if(index > parentBound){
-      				parentBound = index;
-      			}
-      			if(index < childBound){
-      				childBound = index;
-      			}
-      		}
-      		if(useParentBound){
-      			return function(category){
-      				return category.index <= parentBound || category.index >= Constants.sampleCategories.length;
-      			};
-      		} else {
-      			return function(category){
-      				return category.index >= childBound;
-      			}
-      		}
-      	}
-      	
-      	var actions = relationCategories.filter(makeCategoriesFilter(items)).map(function(category) {
+        function makeCategoriesFilter(items) {
+          if (!Constants.isDetailedSample) {
+            return function(category) {
+              if (useParentBound) {
+                return category.index >= Constants.sampleCategories.length;
+              }
+              return category.index > Constants.sampleCategories.length;
+            };
+          }
+          var childBound = Constants.sampleCategories.length - 1;
+          var parentBound = 0;
+          for (sample in items) {
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Add the "let", "const" or "var" keyword to this declaration of "sample" to make it explicit. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2703)